### PR TITLE
Lab2: Reload page on save conflict

### PR DESCRIPTION
**Pull request by molly-moen**
If we receive a 409 error on saving, it means the user has saved elsewhere since loading the page. Our solution to this in our legacy labs is to reload the page, so I reproduced that behavior here. I also added a flag to `ProjectManager` that indicates we initiated a force reload, so the user does not get a confusing pop-up about unsaved changes. We can never save their unsaved changes in this state, so that pop-up is unhelpful.

## Links
- jira ticket: [SL-985](https://codedotorg.atlassian.net/browse/SL-985)

## Testing story
Tested locally that we reload the page on a 409, but not on a normal save or any other save error.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/projects/ProjectManager.ts [hunk 1]
```diff
@@ -286,7 +293,16 @@ export default class ProjectManager {
   private onSaveFail(errorMessage: string, error: Error) {
     this.saveInProgress = false;
     this.executeSaveFailListeners(error);
-    Lab2MetricsReporter.logError(errorMessage, error);
+    if (error.message.includes('409')) {
+      // If this is a conflict, we need to reload the page.
+      // We set forceReloading to true so the client can skip
+      // showing the user a dialog before reload.
+      this.forceReloading = true;
+      reload();
```
Comment (breville) [inline apps/src/lab2/projects/ProjectManager.ts:324]: Can we also log that we are doing a reload?  Hopefully it's safe to do, but there is a potential case where the server continually returns `409` for some other reason, and users experience infinite reloads; we should have a way to be aware of this happening.

Comment (molly-moen) [reply apps/src/lab2/projects/ProjectManager.ts:324]: Makes sense. Do you or @sanchitmalhotra126 have thoughts on whether this should be an info or warning level log?

Comment (sanchitmalhotra126) [reply apps/src/lab2/projects/ProjectManager.ts:324]: Maybe warning, since it's undesirable but recoverable behavior?

**Review by sanchitmalhotra126 — APPROVED** [review]

**Review by breville — APPROVED** [review]

File: apps/src/lab2/projects/ProjectManager.ts (hunk 1 above)
Comment (molly-moen) [reply apps/src/lab2/projects/ProjectManager.ts:324]: done!

**Pull request merged**